### ISSUE TITLE
Layers control should not expand on focus

### ISSUE THREAD

**Issue by Malvoz** [Leaflet/Leaflet#7191]
- [ ] I've looked at the [documentation](http://leafletjs.com/reference.html) to make sure the behavior is documented and expected
- [x] I'm sure this is a Leaflet code issue, not an issue with my own code nor with the framework I'm using (Cordova, Ionic, Angular, React…)
- [x] I've searched through the issues to make sure it's not yet reported

> **Note** <ins>Issue updated per https://github.com/Leaflet/Leaflet/pull/7057/files#r406663620.</ins>

**Steps to reproduce**

1. Visit https://leafletjs.com/examples/layers-control/example.html;
2. <kbd>Tab</kbd> until the layer control is focused;
3. Observe how <s>in Chrome</s> the layers list is expanded<s>, in Firefox it is not expanded</s>.

**Expected behavior**

The layers list should only be expanded after the layer control has been pressed.

I believe this is the original intent, as described in https://github.com/Leaflet/Leaflet/issues/6946#issuecomment-567620007 (emphasis mine): "the default collapsible `L.Control.Layers` can be navigated via keyboard: _**tab-focus on the collapsed layers control, then enter**_, then tab-focus on the set of radio boxes or on the checkboxes."

**Current behavior**

Layer list is expanded <s>in Chrome</s> <ins>(in all browsers)</ins> when the layer control recieves focus.

**Environment**
- Leaflet version: <s>1.6.0, 1.7.1</s> <ins>1.9.2</ins>
- Browser (with version): Firefox 77.0.1 / Chrome 83.0.4103.116
- OS/Platform (with version): Windows 10

**Additional context**

Having the layers list only expand after the layer control has been pressed would also meet WCAG success criterion 2.4.1 Bypass Blocks, the "[Understanding 2.4.1: Bypass Blocks](https://www.w3.org/WAI/WCAG21/Understanding/bypass-blocks.html)" documentation lists several benefits of hiding content (such as a layer list) behind a menu, including:

> - People who use only the keyboard or a keyboard interface can reach content with fewer keystrokes. Otherwise, they might have to make dozens of keystrokes before reaching a link in the main content area. This can take a long time and may cause severe physical pain for some users.

**Comment by IvanSanchez** [comment]
I suspect that Chromium fires some `pointerover` DOM events (or similar) when tabbing, but would like confirmation of this.

A quick&dirty solution for this would be to explicitly expand the layers control on a `focus` event.

**Comment by johnd0e** [comment]
> A quick&dirty solution for this would be to explicitly expand the layers control on a `focus` event.

You can compare tab behavior with https://github.com/Leaflet/Leaflet/pull/7057/files#r406663620

**Comment by Malvoz** [comment]
> A quick&dirty solution for this would be to explicitly expand the layers control on a focus event.

Dirty indeed, that'd negate the benefits of having the menu only expand after button press as described in the "Additional context" section:

> Having the layers list only expand after the layer control has been pressed would also meet WCAG success criterion 2.4.1 Bypass Blocks, the "Understanding 2.4.1: Bypass Blocks" documentation lists several benefits of hiding content (such as a layer list) behind a menu, including:
> 
> > - People who use only the keyboard or a keyboard interface can reach content with fewer keystrokes. Otherwise, they might have to make dozens of keystrokes before reaching a link in the main content area. This can take a long time and may cause severe physical pain for some users.

**Comment by larsgw** [comment]
Should there be a way to _collapse_ the layers control with keyboard navigation?

**Comment by Malvoz** [comment]
@larsgw I think it'd be good if there was a way for _all_ users to be able to close the layers list from the layers list/button itself, I touched on the motivation for that in https://github.com/Leaflet/Leaflet/issues/7526#issuecomment-1158258359, though that may warrant a bit of a redesign of the layers control. However I think that should be discussed separately.
